src/rank_llm/retrieve_and_rerank.py

```diff
@@ -111,68 +144,143 @@ def retrieve_and_rerank(
-            ]
-    print(f"Reranking with {num_passes} passes complete!")
-    for rr in rerank_results:
-        rr.candidates = rr.candidates[:top_k_rerank]
+        model_full_paths = {
+            "rank_zephyr": "castorini/rank_zephyr_7b_v1_full",
+            "rank_vicuna": "castorini/rank_vicuna_7b_v1",
+        }
 
-    # generate trec_eval file & evaluate for named datasets only
-    if isinstance(dataset, str):
-        file_name = reranker.write_rerank_results(
-            retrieval_method.name,
-            rerank_results,
-            shuffle_candidates,
-            top_k_candidates=top_k_retrieve,
-            pass_ct=None if num_passes == 1 else pass_ct,
+        keys_and_defaults = [
+            ("context_size", 4096),
+            ("prompt_mode", PromptMode.RANK_GPT),
+            ("num_few_shot_examples", 0),
+            ("device", "cuda"),
+            ("num_gpus", 1),
+            ("variable_passages", False),
+            ("window_size", 20),
+            ("system_message", None),
+            ("vllm_batched", False),
+        ]
+        [
+            context_size,
+            prompt_mode,
+            num_few_shot_examples,
+            device,
+            num_gpus,
+            variable_passages,
+            window_size,
+            system_message,
+            vllm_batched,
+        ] = extract_kwargs(keys_and_defaults, **kwargs)
+
+        agent = RankListwiseOSLLM(
+            model=model_full_paths[model_path]
+            if model_path in model_full_paths
+            else model_path,
+            name=model_path,
+            context_size=context_size,
+            prompt_mode=prompt_mode,
+            num_few_shot_examples=num_few_shot_examples,
+            device=device,
+            num_gpus=num_gpus,
+            variable_passages=variable_passages,
             window_size=window_size,
-            dataset_name=dataset,
+            system_message=system_message,
+            vllm_batched=vllm_batched,
         )
-        if (
-            dataset in TOPICS
-            and dataset not in ["dl22", "dl22-passage", "news"]
-            and TOPICS[dataset] not in ["dl22", "dl22-passage", "news"]
+
+        print(f"Completed loading {model_path}")
+    elif model_path in ["unspecified", "rank_random", "rank_identity"]:
+        # NULL reranker
+        agent = None
+    else:
+        raise ValueError(f"Unsupported model: {model_path}")
+
+    return agent
+
+
+def extract_kwargs(
+    keys_and_defaults: List[(str, Any)],
+    **kwargs,
+):
+    extracted_kwargs = [
+        kwargs.get(key_and_default[0], key_and_default[-1])
+        for key_and_default in keys_and_defaults
+    ]
+
+    # Check that type of provided kwarg is compatible with the provided default type
+    for i, extracted_kwarg in enumerate(extract_kwargs):
+        if type(keys_and_defaults[i[-1]]) != None and (
+            type(extracted_kwarg) != type(keys_and_defaults[i[-1]])
         ):
-            print("Evaluating:")

```

Comment:
why are you removing these evals
[general tip]: Always aim to separate refactoring from logical changes unless the logical change is the goal of the refactoring (e.g. using a new library that introduces a new dependency, etc).